src/LetsEncrypt/Internal/CertificateFactory.cs

```diff
@@ -183,50 +186,44 @@ private async Task ValidateDomainOwnershipAsync(string domainName, CancellationT
             throw new TimeoutException("Timed out waiting for domain ownership validation.");
         }
 
-        private Exception InvalidAuthorizationError(string domainName, AcmeResult<AuthorizationEntity> authorization)
+        private Exception InvalidAuthorizationError(Authorization authorization)
         {
             var reason = "unknown";
+            var domainName = authorization.Identifier.Value;
             try
             {
-                var errorStub = new { error = new { type = "", detail = "", status = -1 } };
-                var data = JsonConvert.DeserializeAnonymousType(authorization.Json, errorStub);
-                reason = $"{data.error.type}: {data.error.detail}, Code = {data.error.status}";
+                var error = authorization.Challenges.First(a => a.Error != null).Error;
+                reason = $"{error.Type}: {error.Detail}, Code = {error.Status}";
```

Comment:
Can this be updated to log all errors, not just the first? Maybe something like...
```suggestion
                var errors = authorization.Challenges.Where(a => a.Error != null).Select(error => "{error.Type}: {error.Detail}, Code = {error.Status}");
                reason = string.Join(";", errors);
```